exchanges/account/account.go

```diff
@@ -164,29 +166,31 @@ func GetBalance(exch, subAccount string, creds *Credentials, ai asset.Item, c cu
 
 	accounts, ok := service.exchangeAccounts[exch]
 	if !ok {
-		return nil, fmt.Errorf("%s %w", exch, ErrExchangeHoldingsNotFound)
+		return nil, fmt.Errorf("%w for %s", ErrExchangeHoldingsNotFound, exch)
 	}
 
-	subAccounts, ok := accounts.SubAccounts[*creds]
+	subAccounts, ok := accounts.subAccounts[*creds]
 	if !ok {
-		return nil, fmt.Errorf("%s %s %w",
-			exch, creds, errNoCredentialBalances)
+		return nil, fmt.Errorf("%w for %s %s",
+			errNoCredentialBalances, exch, creds)
 	}
 
-	bal, ok := subAccounts[key.SubAccountCurrencyAsset{
+	assets, ok := subAccounts[key.SubAccountAsset{
 		SubAccount: subAccount,
-		Currency:   c.Item,
-		Asset:      ai,
+		Asset:      a,
 	}]
 	if !ok {
-		return nil, fmt.Errorf("%s %s %s %s %w",
-			exch, subAccount, ai, c, errNoExchangeSubAccountBalances)
+		return nil, fmt.Errorf("%w for %s SubAccount `%s` %s %s", errNoExchangeSubAccountBalances, exch, subAccount, a, c)
+	}
+	bal, ok := assets[c.Item]
+	if !ok {
+		return nil, fmt.Errorf("%w for %s SubAccount `%s` %s %s", errNoExchangeSubAccountBalances, exch, subAccount, a, c)
 	}
 	return bal, nil
 }
 
-// Update updates holdings with new account info
-func (s *Service) Update(incoming *Holdings, creds *Credentials) error {
+// Save saves the holdings with new account info
```

Comment:
🚧 Thanks. And sorry, really sorry, to have one more thing still.
I think you missed my point though:
Whatever we're trying to say about this function with it's name, needs to be explicitly documented.
Such as:
```suggestion
// Save saves the holdings with new account info
// incoming should be a full update, and any missing currencies will be zeroed
```